### PR TITLE
Kotlin examples

### PR DESCRIPTION
**Pull request by carlpoole**
This PR adds some Kotlin instructions and examples for Plugin developers.

### CODE REVIEW

**Comment by dwieeb** [comment]
Isn't there some necessary configuration in Android Studio they'd need to do?

**Comment by carlpoole** [comment]
When you add Kotlin code to a project Android Studio prompts you for it like this and you click through a couple of dialogs.
all-modules

File: docs/guide/plugins/android.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ contributors:
 
 # Capacitor Android Plugin Guide
 
-<p class="intro">Building Capacitor plugins for Android involves writing Java to interface with Android SDKs.</p>
+<p class="intro">Building Capacitor plugins for Android involves writing Java or Kotlin to interface with Android SDKs.</p>
```
Comment (dwieeb) [inline docs/guide/plugins/android.md:12]:
```suggestion
<p class="intro">Building Capacitor plugins for Android involves writing Java or <a href="https://developer.android.com/kotlin/overview" target="_blank">Kotlin</a> to interface with Android SDKs.</p>
```

File: docs/guide/plugins/android.md [hunk 2]
```diff
@@ -9,9 +9,9 @@ contributors:
...
 
-<p class="intro">Capacitor embraces standard Android development tools for building Android plugins. We believe that using Java directly will make it easier to use existing solutions on Stack Overflow, share work with existing native developers, and use platform features as soon as they are made available.</p>
+<p class="intro">Capacitor embraces standard Android development tools for building Android plugins. We believe that using Java or Kotlin directly will make it easier to use existing solutions on Stack Overflow, share work with existing native developers, and use platform features as soon as they are made available.</p>
```
Comment (dwieeb) [inline docs/guide/plugins/android.md:14]: Let's just delete this and add a section about Java vs Kotlin later.

```suggestion
```

Comment (carlpoole) [reply docs/guide/plugins/android.md:14]: This paragraph was there before my edit to add Kotlin. Just wanted to make sure it was intended to remove the whole thing before signing off.

**Review by dwieeb — APPROVED** [review]

**Pull request merged**